### ISSUE TITLE
Mutation not sent when optimistic response is present

### ISSUE THREAD

**Issue by mlecoq** [awslabs/aws-mobile-appsync-sdk-js#181]
Hi,

I have a strange behavior in my application. I am using appsync for offline capabilities. Queries work well, but the only mutation I have in the app doesn't work as expected.

Here is the mutation 

` <Mutation mutation={ADD_CONSENT}>
      {createConsent => ( ....
      <Button
                onPress={this.save(createConsent)}
                title="Valider"
                raised
              />
`

I have a save method.

A sum up of it 

`
 save = createConsent => async () => {
    try {
   

      .....

      const consent = {
        mail_breeder: email,
        firstname_breeder: firstname,
        name_breeder: name,
        phone_breeder: phone,
       ....
      };

      await createConsent({
        variables: {
          consent,
        },
        optimisticResponse: {
          __typename: 'Mutation',
          consent: {
            __typename: 'Consent',
            ede_number: consent.ede_number,
            service_id: consent.service_id,
            consent: consent.consent,
            version_cgu,
            date_consent: new Date().toISOString(),
          },
        },
      });

    ....
    } catch (err) {
       ....
      );
    }
  };
`

When I remove optimisticResponse, the mutation is sent (and then I have an error message asking for it). 

With it, mutation is not sent and I have no error message.

Here is how I configured my client: 

```

Amplify.configure({
  Auth: {
    mandatorySignIn: true,
    region: config.REGION,
    userPoolId: config.COGNITO_GROUP_ID,
    identityPoolId: config.COGNITO_IDENTITY_ID,
    userPoolWebClientId: config.COGNITO_CLIENT_ID,
  },
});

const client = new AWSAppSyncClient({
  url: config.APPSYNC_URL,
  region: config.REGION,
  auth: {
    type: AUTH_TYPE.AMAZON_COGNITO_USER_POOLS,
    jwtToken: async () =>
      (await Auth.currentSession()).getAccessToken().getJwtToken(),
  },
  cacheOptions: {
    dataIdFromObject: object => {
      switch (object.__typename) {
        case 'Consent':
          return `consent_${object.service_id}_${object.ede_number}`;
        default:
          return defaultDataIdFromObject(object);
      }
    },
  },
});


```

I don't know how can I go further in analyzing this issue.

**Comment by mlecoq** [comment]
Ok, I've found the issue. In fact, my application was based on redux-offline before adding appsync. I made some customization in config of redux offline. 

While adding appsync, I was not aware that it is based on redux-offline. I have migrated to appsync because I had some reliability issue with redux offline. 

Mutations were blocked by previous operation and after resetting redux offline state, redux offline config (especially in effect method) prevent mutations to be sent.
